File: Programming-Basics-With-JavaScript/Homeworks-And-Labs/06.Nested-Loop/03-additional/09.sumOfTwoNumbers.js
function sumOfTwoNumbers(input) {

    let start = Number(input[0]);
    let end = Number(input[1]);
    let magicNum = Number(input[2]);

    let counter = 0;
    let isFound = false;

    for (let i = start; i <= end; i++) {
        for (let j = start; j <= end; j++) {
            counter++;
            if (i + j == magicNum) {
                isFound = true;
                console.log(`Combination N:${counter} (${i} + ${j} = ${magicNum})`);
            }
            if (isFound) {
                break;
            }
        }
        if (isFound) {
            break;
        }
    }

    if (!isFound) {
        console.log(`${counter} combinations - neither equals ${magicNum}`);
    }
}
sumOfTwoNumbers(['88', '888', '1000']);

File: Python_Basics/Homeworks-And-Labs/06.nested_loops/03.additional/09.sum_of_two_numbers.py
start = int(input())
end = int(input())
magic_num = int(input())

counter = 0
is_found = False

for i in range(start, end + 1):
    for j in range(start, end + 1):
        counter += 1
        if i + j == magic_num:
            is_found = True
            print(f'Combination N:{counter} ({i} + {j} = {magic_num})')
        if is_found:
            break
    
    if is_found:
        break
        
if not is_found:
    print(f'{counter} combinations - neither equals {magic_num}')
